statistics.md

```diff
@@ -0,0 +1,93 @@
+# Collecting user interactions
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Views endpoint
+**POST /api/statistics/views**
+
+To track a page view, make an HTTP POST request to this endpoint.
+
+The sections below describe the parameters for a page view request. All other information about the interaction can be derived from the request itself.
+
+### Parameters
+- `targetId`: The id of the object the user viewed
+- `targetType`: The type of the object the user viewed
+
+### Status codes:
+
+- `201` Created: if the operation succeeded
+- `422` Unprocessable Entity: if any of the parameters are missing or invalid.
+
+### Example item page view:
+
+```json
+{
+  "targetId": "43f9bb3e-f90d-458f-9858-7e4589481d18",
+  "targetType": "item"
+}
+```
+
+
+## Search Endpoint
+**POST /api/statistics/searches**
```

Comment:
I suggest to use usersearches instead than searches as we are focus on user triggered search only (see also general comment to the pull request). Moreover, searches is also very close to search that is a "special" path in our contract

Reply:
I admit, I don't feel strongly here either way, but I'm not I understand why "usersearches" makes the endpoint more meaningful.  How do we know that the search actually came from a user? It could have come from a bot-based client.

Also, if we decide on "usersearches", does that mean we also should have "userviews" (instead of "views")? 

In any case, I don't feel strongly enough to say which one seems more correct. Just noting a few questions that came to mind. I'm fine with whatever others decide.

Reply:
we cannot be sure about how the client will use the endpoint but if the endpoint is named usersearches probably the developer will think a bit more about pushing "system search" event here.

If we want to make an analogy for the views endpoint probably it should be named detailedviews as we don't expect that a developer will record event when the item data are visualized in a list

Reply:
I'm not sure I understand what a "system search" is with regards to tracking Search Statistics.  I had thought that the only search statistics we are tracking are user search statistics anyways (in DSpace 6), so I'm a bit confused about the two types of searches you are referring to.

This discussion makes me wonder if what is missing here is documented, expected use cases for these endpoints. If I'm understanding it correctly, these endpoints should really *only* be used by clients that provide caching of views or searches. That would imply they should never be called directly for *every single view or search* (as that would result in duplicative statistics in scenarios where views and searches are gathered automatically, correct?).

Would better documentation help here, instead of making extra long/descriptive endpoint names ("usersearches" and "detailedviews")? For example, do clients/UIs need to call these endpoints for every view/search?  Or are there very specific scenarios where this is needed.

Reply:
I wouldn't log any views automatically. My goal is to log _all_ views through this endpoint. The rest api shouldn't make assumptions about when something is shown to the user and when it isn't. Just like there are cases where the UI generates multiple views for a single rest request because it was cached, it is conceivable that the UI could trigger more than one GET on a DSO endpoint without it meaning that the user has viewed that DSO's page multiple times. The rest api can't know when something is a page view because it doesn't render pages.

As for the name `usersearches`, I think it will create more confusion than it would solve, as evidenced by the confusion @tdonohue  and I had about it. If `searches` is too similar to `search` I'd go with something more general like `searchevents` and also rename `views` to `viewevents` for consistency.

Reply:
@artlowel : I like the terms `searchevents` and `viewevents` for these endpoints. I think they work well to clarify that we are `POST`ing information about an event that occurred in the UI.  Plus, on the backend these would be logged to our "EventService" (so using the term "event"  is appropriate)

Thanks also for clarifying that all views/searches get logged to this endpoint.  That seems reasonable...I was initially worried that the backend might already be logging these stats (e.g. when a Community/Collection/Item was viewed) and we'd end up with duplication.  But, on second glance, I realize I'm wrong. The only stats I see logged on the backend are Bitstream downloads (which need to be logged there).  Long story short, your approach sounds correct -- all views/searches events should be logged to this endpoint.